### PR TITLE
Changes suggested by clang-tidy: modernize-replace-auto-ptr

### PR DESCRIPTION
**Pull request by quantumsteve**
I made the changes suggested by modernize-replace-auto-ptr as well as removed most other occurences of `std::auto_ptr` I could find and replace with `std::unique_ptr`.

No release notes

testing: code review.

### CODE REVIEW

File: Framework/Algorithms/test/GenerateIPythonNotebookTest.h [hunk 1]
```diff
@@ -149,7 +150,8 @@ class GenerateIPythonNotebookTest : public CxxTest::TestSuite {
     // set up history for the algorithn which is presumably removed from Mantid
     auto ws = API::FrameworkManager::Instance().getWorkspace(wsName);
     API::WorkspaceHistory &history = ws->history();
-    auto pAlg = std::auto_ptr<API::Algorithm>(new NonExistingAlgorithm());
+    auto pAlg = std::unique_ptr<API::Algorithm>(
+        Mantid::Kernel::make_unique<NonExistingAlgorithm>());
```
Comment (SimonHeybrock) [inline Framework/Algorithms/test/GenerateIPythonNotebookTest.h]: `auto pAlg = make_unique`, we don't need the detour.

File: Framework/Algorithms/test/GeneratePythonScriptTest.h [hunk 2]
```diff
@@ -147,7 +148,8 @@ class GeneratePythonScriptTest : public CxxTest::TestSuite {
     // set up history for the algorithn which is presumably removed from Mantid
     auto ws = API::FrameworkManager::Instance().getWorkspace(wsName);
     API::WorkspaceHistory &history = ws->history();
-    auto pAlg = std::auto_ptr<API::Algorithm>(new NonExistingAlgorithm());
+    auto pAlg = std::unique_ptr<API::Algorithm>(
+        Mantid::Kernel::make_unique<NonExistingAlgorithm>());
```
Comment (SimonHeybrock) [inline Framework/Algorithms/test/GeneratePythonScriptTest.h]: As above, use ony `make_unique` without extra constructor.

**Comment by SimonHeybrock** [comment]
No more comments. I think all changes look good. Could be merged after addressing the 2 things I mentioned above.

**Comment by quantumsteve** [comment]
@SimonHeybrock I made the two changes you suggested.

**Pull request merged**